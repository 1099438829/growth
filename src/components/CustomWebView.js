import React, { Component, PropTypes } from 'react';
import { WebView, Dimensions } from 'react-native';
import HtmlHelper from '../utils/HtmlHelper';

class CustomWebView extends Component {
  static componentName = 'CustomWebView';

  static propTypes = {
    html: PropTypes.string,
  };

  static defaultProps = {
    html: '',
  };

  constructor(props) {
    super(props);
    this.state = {};
    this.webview = null;
  }

  render() {
    return (
      <WebView
        ref={(webview) => {
          this.webview = webview;
        }}
        scalesPageToFit
        startInLoadingState
        source={{ html: HtmlHelper.getHtml(this.props.html) }}
        style={{ height: Dimensions.get('window').height, backgroundColor: 'white' }}
        onNavigationStateChange={this.onNavigationStateChange}
        injectedJavaScript=""
      />);
  }
}
export default CustomWebView;
