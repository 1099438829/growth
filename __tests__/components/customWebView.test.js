import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import CustomWebView from '../../src/components/CustomWebView';

it('renders correctly', () => {
  // eslint-disable-next-line no-unused-vars
  const customWebView = renderer.create(
    <CustomWebView value={'<h1>233333</h1>'} />,
  );

  const customWebViewJson = customWebView.toJSON();
  expect(customWebViewJson).toMatchSnapshot();
});
