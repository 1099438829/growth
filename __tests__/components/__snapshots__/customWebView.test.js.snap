// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTWebView
    allowsInlineMediaPlayback={undefined}
    automaticallyAdjustContentInsets={undefined}
    bounces={undefined}
    contentInset={undefined}
    dataDetectorTypes={undefined}
    decelerationRate={undefined}
    injectedJavaScript=""
    mediaPlaybackRequiresUserAction={undefined}
    messagingEnabled={false}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={undefined}
    scalesPageToFit={true}
    scrollEnabled={undefined}
    source={
      Object {
        "html": "",
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "backgroundColor": "white",
          "height": 1334,
        },
        Object {
          "flex": 0,
          "height": 0,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255,255,255,0.8)",
        "flex": 1,
        "height": 100,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;
